gstreamer-pbutils/src/auto/discoverer.rs

```diff
@@ -0,0 +1,184 @@
+// This file was generated by gir (d933f9a+) from gir-files (???)
+// DO NOT EDIT
+
+use DiscovererInfo;
+use Error;
+use ffi;
+use glib;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use gst_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct Discoverer(Object<ffi::GstDiscoverer, ffi::GstDiscovererClass>);
+
+    match fn {
+        get_type => || ffi::gst_discoverer_get_type(),
+    }
+}
+
+impl Discoverer {
+    pub fn new(timeout: gst::ClockTime) -> Result<Discoverer, Error> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::gst_discoverer_new(timeout.to_glib(), &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+}
+
+unsafe impl Send for Discoverer {}
+unsafe impl Sync for Discoverer {}
+
+pub trait DiscovererExt {
+    fn discover_uri(&self, uri: &str) -> Result<DiscovererInfo, Error>;
+
+    fn discover_uri_async(&self, uri: &str) -> bool;
```

Comment:
This bool should be a Result, gir can do that automatically for you. Check e.g. gst_element_link()

Reply:
It actually always returns true, except when discoverer pointer would not be a discoverer.

Reply:
Ok so it can never return false actually and is useless :)